### ISSUE TITLE
V4.0 Bug Bash

### ISSUE THREAD

**Issue by Cooldude2606** [explosivegaming/scenario#85]
- [x] Admin buttons in player list dont work
- [x] Report button is labled as admin commands
- [x] Reports aren't made
- [x] Sync not updating
- [x] Jail still clears inventory

**Comment by aldldl** [comment]
I did it wrong - Warnings not working correctly

**Comment by aldldl** [comment]
Mods dont have report button but Dev does

**Comment by aldldl** [comment]
[] the discord sycn on /promote admin works but is doubled each time

**Comment by Cooldude2606** [comment]
```
[FSM] Assigning Roles: "train" to: ookl
[FSM] Event Failed: "ExpGamingCore.Role@^4.0.0/on_role_change" (.../currently-playing/modules/ExpGamingCore/Role/config.lua:2: attempt to index field 'character' (a nil value))```

**Comment by Cooldude2606** [comment]
- [ ] Damage value are not floored

**Comment by Cooldude2606** [comment]
- [ ] Jail does not work, nil value for Admin.jail

**Comment by aldldl** [comment]
- [] /tag-clear aldldl Wont work because he is a higher rank than myself
- [] /tag-clear self Wont work because I am a higher rank than myself

**Comment by aldldl** [comment]
- [ ] /temp-ban {greifernamehere} said the guy is temp-banned but he could still grief (No error)

**Comment by Cooldude2606** [comment]
> 
> 
>     * [ ]  I did it wrong - Warnings not working correctly

@aldldl could you please give more information on this; which part does not work.

**Comment by aldldl** [comment]
/warn didn't do anything. Also I think it didn't auto ban but I can't remember if that was also the case.

**Comment by Cooldude2606** [comment]
All known bugs fixed.

**Comment by aldldl** [comment]
- []  Spawn issue with tones of warps/labels is back in V4
- []  Some really weird stuff with Dark's rank and a jail issue that was happening (see logs)
- []   Dark's inventory got cleared somehow and we dont know why (Does Temp ban clear inventory?)
- []  Can't kill a jailed player
- []  Jailed players in cars/trains etc are unjailable
- []  a few V3.6 items back as issues
--- []  Warp sorts are case sensitive again
- []  Spawn should be spawn once that is fixed for easy /home ing

**Comment by aldldl** [comment]
[ ] discord sync broken again for reports.

**Comment by flbuddymooreiv** [comment]
- [ ] Maximum Homes count mismatch
- [ ] Home Remove output message context is wrong: "Invalid" followed by "Removed"
- [ ] Your home 'xyz' _**as**_ been removed

![image](https://user-images.githubusercontent.com/3165630/51580684-5d8c1c80-1e93-11e9-8ea3-a9537797a8ca.png)

**Comment by flbuddymooreiv** [comment]
I believe I just set a few homes and when adding a third (or fourth?) I got
that error. I will try to create from scratch at some point!

**Comment by flbuddymooreiv** [comment]
>     * [ ]   Spawn issue with tones of warps/labels is back in V4

`/c for k,v in pairs(game.player.force.find_chart_tags(game.player.surface)) do if v.text == 'Warp: Spawn' then v.destroy() end end`

**Comment by flbuddymooreiv** [comment]
>     * [ ]   Spawn issue with tones of warps/labels is back in V4

I believe this line and the `valid` field may have been the original issue. But I can verify that the whole list of warps gets recreated every time a new warp gets created, so hopefully this helps:

https://github.com/explosivegaming/scenario/blob/c31bca9dba65c3738e9556d0a6c3708188b2016d/modules/WarpPoints/control.lua#L165

**Comment by Cooldude2606** [comment]
>  Can't kill a jailed player
Why do you need to kill a jailed player, as i would not consider this a bug.

**Comment by Cooldude2606** [comment]
>  Maximum Homes count mismatch
>  Corollary to above: Negative Homes count
Above two should be fixed when i fixed the duel return however this is not tested.

**Comment by Cooldude2606** [comment]
> Some really weird stuff with Dark's rank and a jail issue that was happening (see logs)

Will need some more info on this one @aldldl

**Comment by Cooldude2606** [comment]
Unable to reproduce: Some really weird stuff with Dark's rank and a jail issue that was happening (see logs)

Closeing issue as all other bugs have been fixed see [v4.0.1](https://github.com/explosivegaming/scenario/commit/be3cb7d0d9fe6b85af7913f4327630ff21a273ff)

**Comment by aldldl** [comment]
It removed good jail rank, but he still couldn't move./clear-all and /unjail. I have not reproduced it again when banning or jailing sense.
